yarn-project/archiver/src/archiver/l1/validate_trace.ts

```diff
@@ -0,0 +1,219 @@
+    debugTraceOld: false,
+    traceTransactionOld: false,
+  };
+
+  try {
+    // Get the latest block
+    let latestBlock = await client.getBlock({ blockTag: 'latest' });
+
+    // Loop back to find a block with transactions (handles dev/test scenarios)
+    while (
+      (!latestBlock.transactions || latestBlock.transactions.length === 0) &&
+      latestBlock.number &&
+      latestBlock.number > 0n
+    ) {
+      logger.debug(`Block ${latestBlock.number} has no transactions, checking previous block`);
+      latestBlock = await client.getBlock({ blockNumber: latestBlock.number - 1n });
+    }
+
+    if (!latestBlock.transactions || latestBlock.transactions.length === 0) {
+      logger.warn('No blocks with transactions found from latest back to genesis, cannot validate trace availability');
+      return result;
+    }
+
+    // Get a transaction from the found block
+    const recentTxHash = latestBlock.transactions[0] as Hex;
+
+    // Test debug_traceTransaction with recent block
+    result.debugTraceRecent = await testTraceMethod(
+      client,
+      recentTxHash,
+      callTraceSchema,
+      'debug_traceTransaction',
+      'recent',
+    );
+
+    // Test trace_transaction with recent block
+    result.traceTransactionRecent = await testTraceMethod(
+      client,
+      recentTxHash,
+      traceTransactionResponseSchema,
+      'trace_transaction',
+      'recent',
+    );
+
+    // Get a block from 512 blocks ago
+    const oldBlockNumber = latestBlock.number ? latestBlock.number - 512n : null;
+    if (!oldBlockNumber || oldBlockNumber < 0n) {
+      logger.debug('Cannot test old blocks, blockchain too short');
+      return result;
+    }
+
+    let oldBlock = await client.getBlock({ blockNumber: oldBlockNumber });
+
+    // Loop back to find a block with transactions (handles dev/test scenarios)
+    while ((!oldBlock.transactions || oldBlock.transactions.length === 0) && oldBlock.number && oldBlock.number > 0n) {
+      logger.debug(`Block ${oldBlock.number} has no transactions, checking previous block`);
+      oldBlock = await client.getBlock({ blockNumber: oldBlock.number - 1n });
+    }
+
+    if (!oldBlock.transactions || oldBlock.transactions.length === 0) {
+      logger.debug(
+        'No blocks with transactions found from old block back to genesis, cannot validate trace availability for old blocks',
+      );
+      return result;
+    }
+
+    const oldTxHash = oldBlock.transactions[0] as Hex;
+
+    // Test debug_traceTransaction with old block
+    result.debugTraceOld = await testTraceMethod(client, oldTxHash, callTraceSchema, 'debug_traceTransaction', 'old');
+
+    // Test trace_transaction with old block
+    result.traceTransactionOld = await testTraceMethod(
+      client,
+      oldTxHash,
+      traceTransactionResponseSchema,
+      'trace_transaction',
+      'old',
+    );
+  } catch (error) {
+    logger.error(`Error during trace availability validation: ${error}`);
```

Comment:
What do you think of only using .error() for terminating errors and warn for errors that got swallowed/we can continue?

```suggestion
    logger.warn(`Error during trace availability validation: ${error}`);
```